lms/views/predicates/_predicates.py

```diff
@@ -0,0 +1,101 @@
+def is_canvas_file(_context, request):
+    """Get if this is a Canvas file launch."""
+
+    return "canvas_file" in request.params
+
+
+def is_vitalsource_book(_context, request):
+    """Get if this is a Vitalsource launch."""
+
+    return "vitalsource_book" in request.params
+
+
+def is_deep_linking_configured(_context, request):
+    """
+    Get if this is a deep-linked launch.
+
+    This is where the document is stored in the LMS and provided to us at
+    launch time in the params. This is here because we stored it there when we
+    sent the deep linking configuration.
+    """
+
+    return "url" in request.params
+
+
+def is_db_configured(context, request):
+    """Get if this launch has assignment configuration in the DB."""
+
+    return _is_assignment_based(context, request, context.resource_link_id)
+
+
+class ResourceLinkParams:
```

Comment:
`HistoryResourceLinkParams`?

Reply:
... _wait for it!_

I'm intending to add `resource_link_id` to this in an upcoming PR